impeller/entity/entity_pass.cc

```diff
@@ -901,9 +901,13 @@ bool EntityPass::OnRender(
   }
   auto clear_color_size = pass_target.GetRenderTarget().GetRenderTargetSize();
 
-  if (!collapsed_parent_pass && GetClearColor(clear_color_size).has_value()) {
+  if (!collapsed_parent_pass && (GetClearColor(clear_color_size).has_value() ||
+                                 renderer.GetContext()->GetBackendType() ==
+                                     Context::BackendType::kVulkan)) {
```

Comment:
I understand the concern about branches but adding a virtual method that is only implemented in one backend is roughly equivalent to a branch, no?.

I could definitely make this clearer by pulling the condition out to a helper method though

Reply:
(If you don't like this one, you're gonna _hate_ the next PR I send you)

Reply:
> I understand the concern about branches but adding a virtual method that is only implemented in one backend is roughly equivalent to a branch, no?.

There is a subtle difference that when you make a virtual method.  You are contributing to an interface which collectively defines the abstraction.  If you add branching statements there is no single place that defines or enforces the interface.

Imagine we wanted to add a directx backend, if we had the branch we wouldn't know that we need to make a decision about wether we need to setup the renderpass regardless, like we do in vulkan.

Also, when we expand the abstraction, people can read through that interface and understand the system hierarchically.  With branches the interface exists but it is implicit and spread around multiple locations.

Reply:
I don't think exposing ImageLayout is the right way to do that though, that requires tons of Vulkan only internal state to be exposed to entity pass.

> Imagine we wanted to add a directx backend,

I mean, we can handle that if/when we do DirectX.

Reply:
> I don't think exposing ImageLayout is the right way to do that though, that requires tons of Vulkan only internal state to be exposed to entity pass.

It won't be tons of state, it will just be a hook like `PrepareRenderPass()`.  You understand how branching on the backend type compromises the abstraction, right?

Would you like me to take a stab at proposing an alternative that expands the interface?  If this comes up in a later PR it could potentially serve as a template there too.

Reply:
I think the abstraction being intentionally leaky (as its private API) is valuable.

Reply:
If you wanna take a crack at it, have at it, I'm not in a rush :)